pkg/runtime/runtime_test.go

```diff
@@ -21,17 +20,17 @@ func TestAutoMaxProcs(t *testing.T) {
 	runtime.GOMAXPROCS(2)
 	assert.Equal(t, 2, runtime.GOMAXPROCS(0))
 
-	os.Setenv("GOMAXPROCS", "1000m")
+	t.Setenv("GOMAXPROCS", "1000m")
 	// set new limit
 	SetMaxProcs()
 	assert.Equal(t, 1, runtime.GOMAXPROCS(0))
 
-	os.Setenv("GOMAXPROCS", "1500m")
+	t.Setenv("GOMAXPROCS", "1500m")
 	// set new limit
 	SetMaxProcs()
 	assert.Equal(t, 1, runtime.GOMAXPROCS(0))
 
-	os.Setenv("GOMAXPROCS", "2000m")
```

Comment:
Did you verify that this correctly handles multiple calls for the same variable?  The docs don't say anything about it, but if you imagine it's calling `t.Cleanup` internally, then it's probably OK as cleanups run in the reverse of the order in which `t.Cleanup` was called.

Reply:
From docs:
> Setenv calls os.Setenv(key, value) and uses Cleanup to restore the environment variable to its original value after the test.

Reply:
I guess you don't have to imagine.  So, yes, that seems to imply that it's safe to use multiple times in a test.

Reply:
> Did you verify that this correctly handles multiple calls for the same variable? The docs don't say anything about it, but if you imagine it's calling `t.Cleanup` internally, then it's probably OK as cleanups run in the reverse of the order in which `t.Cleanup` was called.

@djmitche I've updated `TestAutoMaxProcs` into table test. This should makes it clearer :smiley: